mne/io/fiff/tests/test_raw.py

```diff
@@ -115,6 +115,33 @@ def test_copy_append():
     assert_equal(data.shape[1], 2 * raw._data.shape[1])
 
 
+@testing.requires_testing_data
+def test_append_channels():
+    """Test raw splitting / re-appending channel types
+    """
+    raw = Raw(fif_fname, preload=True)
```

Comment:
> what happens if I try to channel-concatenate raws with different numbers of time points? 
```python
if not all([self.n_times == raw.n_times for raw in raw_list]):
            raise AssertionError('Data have unequal numbers of times')
```
this is already in there and does it, no?

>what happens if I try to channel-concatenate raws with different sampling rates, but the same number of time points?

_merge_info raises a decent error if the sampling frequencies don't match up

> what happens if I try to channel-concatenate raws with .info with e.g. different subject names?

Right now it does not raise any errors for this kind of discrepancy, but this should be the job of `_merge_info` rather than the `append_channels` function, no?